### PR TITLE
Feature/simplify api backends collection

### LINKED ISSUE #1091: Design API backend schema
In order to decouple API backends from Api Umbrella, we will need a collection for API backend that is separate from the proxy settings. In other words, the schema needs to be changed and separated from proxies and their settings.

Idea: Align with Swagger specification or Open API specification (depending on the schedule of the latter). This might enable importing API backend configuration by importing Swagger (or Open API) file. Only after adding the API backend, the owner would assign a proxy to it.

Potential user story related to the Swagger idea:
>  As an API owner
I want to add my API by importing a Swagger file
so that I can reuse my Swagger file and do not need to type in the API details

As a minimum requirement, all fields that exist both in our collection and in Swagger spec should match.

# Definition of done
* [ ] Api Backend schema exists 
* [ ] Api Backend does not contain API Umbrella related settings
* [ ] When applicable, schema field(s) confirm to Swagger / Open API specification

### LINKED ISSUE #1379: Update Add API wizard
Update Add API wizard to add an API without proxy settings. (Proxy should be added - or in the future selected - in separate UI.) This UI will add an API with basic details to Catalog.

# Definition of done
* [x] User can enter all details related purely to  API backend such as name, description, base URL with Add API wizard 
  * [x] Add API wizard does not contain any details or settings related to proxy or selecting proxy
* [x] When user presses Publish API backend, the API will be added to API backends collection
  * [x] User get redirected to single API view for the newly created API

### PR DESCRIPTION
**Pull request by brylie**
Closes #1091 
Closes #1379 

# Changes
- Rename `ApiBackends` collection to `Apis`
  - rename file to `index.js` for clean imports
- Remove API Umbrella fields from Apis schema
- Create Add API form
- Remove Add API Wizard
- Remove Initial Setup Wizard
- Edit basic settings/branding check to display modal dialogue to admin
- Cleanup unused packages
- Cleanup/rename/reorganize files along the way
- fix bugs related to user profile and user menu

### CODE REVIEW

**Comment by brylie** [comment]
@apinf/core please review

**Comment by bajiat** [comment]
@frenchbread Have you started on the review yet? @brylie is waiting for your feedback and other tasks are blocked by or at least not progressing because of this

**Comment by bajiat** [comment]
Assigned to @jykae since he is available for review

**Comment by jykae** [comment]
Reviewing

**Comment by brylie** [comment]
@Alapan please be aware that we are deprecating the initial setup wizard as part of this pull request. We decided the initial setup wizard was no longer necessary, since API Umbrella settings are no longer necessary when starting the project.

The overall goal of this pull request is to decouple from API Umbrella, so we have removed a lot of code in the process.

**Comment by jykae** [comment]
@brylie Modal about missing Branding & Settings is jumping whatever page I go when testing with fresh db. I think that none of Branding or Settings are actually needed. Could it be shown only once when admin first signs up if we are serious to remove setup wizard? I would keep the setup wizard though as it is useful guide through for setting Branding & Settings fields.

Also after setting Branding, modal is not jumping but noticed this:
nayttokuva 2016-08-19 kello 15 46 28

I seem to have admin permissions, but on the right side it looks like I am not logged in and don't have links. On front page that shows correctly that I am logged in as admin.

**Comment by jykae** [comment]
Pushed fix for user menu not showing.

File: setup/server/methods.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Meteor.methods({
+  "isInitialSetupComplete": function() {
```
Comment (frenchbread) [inline setup/server/methods.js]: Please change this line to ES6 like syntax:
```js
Meteor.methods({
    isInitialSetupComplete () {
        // ...
    }
});

**Comment by frenchbread** [comment]
One more note: as we discussed, instead of naming file for `Apis` collection with the same name as its parent folder, its better to name file as `index.js` so that the path to it would be `import { Apis } from '/apis/collection'`.

**Comment by frenchbread** [comment]
While testing got an error: 
```
Exception while invoking method 'getApiStatus' TypeError: Cannot read property 'statusCode' of undefined
```

and probably because of that dashboard is not loading any data.

**Comment by brylie** [comment]
>While testing got an error:
>
>Exception while invoking method 'getApiStatus' TypeError: Cannot read property 'statusCode' of undefined
>
> and probably because of that dashboard is not loading any data.


Make sure you reset your Meteor database. I made a change to the field name/function, and any previously saved API Backend may have the old `baseUrl` field, as opposed to the new `url` field.

**Comment by jykae** [comment]
Hey @brylie @frenchbread 
I made yesterday subset of this PR, that has only Apis schema & add api form changes, and necessary Apis collection bindings to keep existing features functional. 

Imho it would be easier first step than trying whole bunch of cleanups, restructuring etc. in one PR. Let's put work in pieces so that PRs won't grow too large to follow, it makes debugging harder.

**Comment by brylie** [comment]
I have made the suggested changes
- use ES6 / ES2015 syntax for methods
- use index.js for collection filename
- refactor import statements using /apis/collection or relative imports
- fix user menu
- fix user profile

@apinf/core please review.

**Comment by jykae** [comment]
🆒 I checkout asap 👍

**Comment by brylie** [comment]
Kiitos. I will resolve conflicts.

**Comment by jykae** [comment]
Quite well done. But one undefined,
```
Exception in template helper: TypeError: Cannot read property 'apiLogoFileId' of undefined
    at Object.uploadedApiLogoLink (http://localhost:3000/app/app.js?hash=96e4a056f39075ebf957bfa580142e6cfa6444fb:3916:69)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2994:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:1653:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3046:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3687:12)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3045:27
    at Spacebars.call (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:172:18)
    at Spacebars.mustacheImpl (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:106:25)
    at Object.Spacebars.mustache (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:110:39)
    at src (http://localhost:3000/app/app.js?hash=96e4a056f39075ebf957bfa580142e6cfa6444fb:3880:26)
```
That appears to come from "apis/logo/view/client/view.js" line 16

**Comment by jykae** [comment]
On frontpage, missing username only underscore visible:
nayttokuva 2016-08-23 kello 14 40 40

After these and conflict fixes I am fine with this for merging.
If there are plans to remove frontpage or change layout completely you can skip this frontpage bug.

**Comment by brylie** [comment]
I would like to leave the `home`template alone in this PR. This is because I am switching to using the master layout for the home page, in a separate PR, and want to avoid conflicts with that work.

**Comment by brylie** [comment]
Temporarily restoring the branch, so I can replicate changes.

**Pull request merged**